### PR TITLE
search: Fix backspace item deletion bug

### LINKED ISSUE #32352: Character deletion deletes notepad item when selected

### PR DESCRIPTION
**Pull request by fkling**
Fixes #32352 

This was caused by the fact that I switched from listing to key *up*
events to key *down* events, but didn't make the same change to the
event handler bound to the textarea, so all key events bubbled up to the
parent.

Instead of relying on stopping the event propagation, the list event
handler now checks whether an annotation textarea is focused and if yes
it ignores the event.

This commit also adds the ability to remove focus from the annotation
textarea by pressing Escape.

## Test plan

With enabled search stack feature:

- Navigate to any search, open/expand the notepad widget and add the current search. If the current search is not automatically selected, select it.
- Open the selected item's annotation input (if not already open) and enter any text.
- Press backspace or delete to delete text. The selected item does not get deleted.
- Press Escape, the textarea looses focus

### CODE REVIEW

**Review by limitedmage — COMMENTED** [review]
Question: should Backspace even delete items? In the browser, Backspace usually means navigating back (at least on Windows); we would be overriding this default behavior. On Notebooks, we're using `Cmd+Backspace` (for Mac which may not have a dedicated delete key) and `Delete` to delete a block.

**Comment by fkling** [comment]
@limitedmage Good point. Will remove backspace for now.

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/SearchStack.tsx [hunk 1]
```diff
@@ -210,7 +210,6 @@ export const SearchStack: React.FunctionComponent<SearchStackProps> = ({ initial
                     break
                 // Delete selected entries
                 case 'Delete':
-                case 'Backspace':
```
[the remark this answers is not in the document]
Comment (benvenker) [reply client/web/src/search/SearchStack.tsx:208]: *but I don't think that needs to block this from merging

[the remark this answers is not in the document]
Comment (fkling) [reply client/web/src/search/SearchStack.tsx:208]: Added support for CMD+backspace in the last commit.

**Pull request merged**